test/e2e/framework/framework.go

```diff
@@ -230,7 +198,7 @@ func (f *Framework) matchNginxConditions(name string, matcher func(cfg string) b
 
 		o, err := f.ExecCommand(pod, cmd)
 		if err != nil {
-			return false, err
+			return false, nil
```

Comment:
why are we not returning the `err` anymore?

Reply:
Because returning an error breaks the poll loop

Reply:
oops - so that means all the polling e2e tests have been doing weren't really polling?!